### PR TITLE
chore: update tidb addon to kb 1.0 api

### PR DESCRIPTION
**Pull request by cjc7373**

### CODE REVIEW

**Comment by cjc7373** [comment]
@Y-Rookie PTAL

**Review by zjx20 — APPROVED** [review]

File: addons/tidb/templates/_helpers.tpl [hunk 1]
```diff
@@ -77,3 +84,28 @@ tikv-config-constraints
...
+
+{{- define "tidb.tikv7.componentDefNamePrefix" -}}
+tikv-7-
+{{- end -}}
+
+{{- define "tidb.tikv7.compDefName" -}}
+{{ include "tidb.tikv7.componentDefNamePrefix" . }}{{ .Chart.Version }}
+{{- end -}}
+
+{{- define "tidb.tidb7.componentDefNamePrefix" -}}
```
Comment (Y-Rookie) [inline addons/tidb/templates/_helpers.tpl:104]: better to use regexp pattern `^tidb-7-` instead name prefix

File: addons/tidb/templates/componentdefinition-tidb.yaml [hunk 2]
```diff
@@ -10,23 +12,40 @@ spec:
...
+    - name: TIDB_COMPONENT_NAME
+      valueFrom:
+        componentVarRef:
+          optional: false
+          componentName: Required
     - name: PD_HOST
       valueFrom:
         serviceVarRef:
-          compDef: tidb-pd-7
+          compDef: {{ include "tidb.pd7.compDefName" . }}
```
Comment (Y-Rookie) [inline addons/tidb/templates/componentdefinition-tidb.yaml:29]: check if it's necessary to replace this with a regexp pattern match

File: addons/tidb/templates/_helpers.tpl [hunk 3]
```diff
@@ -77,3 +84,28 @@ tikv-config-constraints
 {{- define "tidb.pd.configConstraintName" -}}
 tidb-pd-config-constraints
 {{- end -}}
+
+{{- define "tidb.pd7.componentDefNamePrefix" -}}
+tidb-pd-7-
```
Comment (Y-Rookie) [inline addons/tidb/templates/_helpers.tpl:89]: The references to compDef in backupPolicyTemplate need to be modified, please check.

**Review by Y-Rookie — APPROVED** [review]

**Review by zjx20 — APPROVED** [review]

**Pull request merged**